internal/server/v2/observation/calculation_util.go

```diff
@@ -300,3 +300,36 @@ func evalBinaryExpr(
 	}
 	return result, nil
 }
+
+// Recursively iterate through the AST and perform the calculation.
+func evalExpr(
+	node ast.Node,
+	formula *VariableFormula,
+) (map[string]map[string][]*pb.PointStat, error) {
```

Comment:
Instead of define a nested function `computeChildObs`, maybe just call it out in line 321 and line 325 as it's only 4 lines. I feel that makes the switch part easier to follow.

Reply:
Good idea, done